src/components/equipment/BarcodeScanner.tsx

```
@@ -0,0 +1,178 @@
+    const handleSubmit = async () => {
+        if (!barcode) return;
+        setIsLoading(true);
+        setResult(null);
+
+        if (mode === "check_out") {
+            if (!reservationId) {
+                setResult({ success: false, message: "Reservation ID is required for checkout." });
+                setIsLoading(false);
+                return;
+            }
+            const res = await checkOutEquipment(barcode, reservationId);
```

ESLint

[error] Replace `············` with `······`

(prettier/prettier)